README.md

```diff
@@ -598,54 +598,55 @@ meant to be able to change with it.
-    after future refactoring.
+    Prefer using verifying doubles over normal doubles.
     <sup>[[link](#doubles)]</sup>
 
-    This generally means you should use them with more isolated/behavioral
-    tests rather than with integration tests.
+    Verifying doubles are a stricter alternative to normal doubles that
+    provide guarantees, e.g. a failure will be triggered if an invalid
+    method is being stubbed or a method is called with an invalid number
+    of arguments.
+
+    In general, use doubles with more isolated/behavioral tests rather
+    than with integration tests.
+
+    *NOTE*: There is no justification for turning
+    `verify_partial_doubles` configuration option off. That will
+    significantly reduce the confidence in partial doubles.
 
     ```ruby
-    # double an object
-    article = double('article')
+    # good - verifying instance double
+    article = instance_double('Article')
+    expect(article).to receive(:author).and_return(nil)
+
+    presenter = described_class.new(article)
+    expect(presenter.title).to include('by an unknown author')
+
+
+    # good - verifying object double
+    article = object_double(Article.new, valid?: true)
+    expect(article.save).to be true
 
-    # stubbing a method
+
+    # good - verifying partial double
     allow(Article).to receive(:find).with(5).and_return(article)
+
+
+    # good - verifying class double
+    notifier = class_double('Notifier')
+    expect(notifier).to receive(:notify).with('suspended as')
     ```
 
-    *NOTE*: if you stub a method that could give a false-positive test result, you
+    *NOTE*: If you stub a method that could give a false-positive test result, you
     have gone too far. See below:
 
     ```ruby
     # bad
-    subject { double('article') }
-
-    describe '#summary' do
-      context 'when summary is not present' do
-        # This stubbing of the #nil? method, makes the test pass, but
-        # you are no longer testing the functionality of the code,
-        # you are testing the functionality of the test suite.
-        # This test would pass if there was not a single line of code
-        # written for the Article class.
-        it 'returns nil' do
-          summary = double('summary')
-          allow(subject).to receive(:summary).and_return(summary)
-          allow(summary).to receive(:nil?).and_return(true)
-          expect(subject.summary).to be_nil
-        end
-      end
-    end
-
-    # good
-    subject { double('article') }
+    subject { instance_double('Article') }
 
     describe '#summary' do
       context 'when summary is not present' do
-        # This is no longer stubbing all of the functionality, and will
-        # actually test the objects handling of the methods return value.
+        # This is not testing any production code anymore, an
+        # expectation is made directly on the stubbed method.
         it 'returns nil' do
           allow(subject).to receive(:summary).and_return(nil)
```

Comment:
We discourage stubbing the subject under test (as in #51) so there’s more than one problem with this example 😁

Reply:
I knew this will come up, and that makes it harder to come up with an example for this one if you follow the other rules.

Do you have a good example of over-stubbing in mind?
I'm inclined to remove this note along with the example.

Reply:
> Do you have a good example of over-stubbing in mind?

Sorry, I forgot to reply. But no, I don’t have any better examples in mind.